src/_pytest/unittest.py

```diff
@@ -217,6 +220,7 @@ def wrapped_testMethod(*args, **kwargs):
                 expecting_failure = self._expecting_failure(testMethod)
                 if expecting_failure:
                     raise
+                self._needs_explicit_tearDown = True
                 raise _GetOutOf_testPartExecutor(exc)
 
         setattr(self._testcase, self._testcase._testMethodName, wrapped_testMethod)
```

Comment:
Does it pass https://github.com/pytest-dev/pytest/pull/5996#discussion_r345282784 then?

Reply:
Yep! 👍